src/meltano/core/plugin/superset.py

```diff
@@ -0,0 +1,168 @@
+"""Plugin glue code for Superset."""
+import logging
+import subprocess
+from typing import List
+
+import structlog
+
+from meltano.core.behavior.hookable import hook
+from meltano.core.error import AsyncSubprocessError
+from meltano.core.plugin.error import PluginExecutionError
+from meltano.core.plugin_invoker import PluginInvoker
+from meltano.core.setting_definition import SettingDefinition
+
+from . import BasePlugin, PluginType
+
+logger = structlog.getLogger(__name__)
+
+
+class SupersetInvoker(PluginInvoker):
+    """Invoker that prepares env for Superset."""
+
+    def env(self):
+        """Environment variables for Superset.
+
+        Returns:
+            Dictionary of environment variables.
+        """
+        env = super().env()
+
+        env["SUPERSET_HOME"] = str(self.plugin_config_service.run_dir)
+        env["SUPERSET_CONFIG_PATH"] = str(self.files["config"])
+        env["FLASK_APP"] = "superset"
+
+        return env
+
+
+class Superset(BasePlugin):
+    """Plugin glue code for Superset."""
+
+    __plugin_type__ = PluginType.UTILITIES
+
+    invoker_class = SupersetInvoker
+
+    EXTRA_SETTINGS = [SettingDefinition(name="_config_path")]
+
+    @property
+    def config_files(self):
+        """Return the configuration files required by the plugin.
+
+        Returns:
+            Dictionary of config file identifiers and filenames
+        """
+        return {"config": "superset_config.py"}
+
+    @hook("before_configure")
+    async def before_configure(self, invoker: SupersetInvoker, session):  # noqa: WPS217
+        """Write plugin configuration to superset_config.py.
+
+        Args:
+            invoker: the active PluginInvoker
+            session: metadata database session
+
+        Raises:
+            PluginExecutionError: if config file couldn't be found
+        """
+        config = invoker.plugin_config_processed
+
+        config_script_lines = [
+            "import sys",
+            "module = sys.modules[__name__]",
+            f"config = {str(config)}",
```

Comment:
Can we get these dynamically from env vars and avoid secrets being stored in clear text?

Reply:
@aaronsteers I considered that, but env vars are all strings and `superset_config.py` expects values to have their proper types, e.g. integer. To read from env vars and convert to proper types, we'd need to hard-code the type of every setting, which is not reasonable and out of scope.

Reply:
@aaronsteers I see your concern, though. Do you know if we have a dedicated issue for that as it's also a problem with Singer and Airflow?

Reply:
It's a few more lines of code, but since this code path does have access to type info, we could have two declarations - one for secrets (kind=password) and the others in clear text, as is now.

We could also just encode conversions for all variables, but that's not going to be possible in this timeline.

Let's log as a follow-on issue.

Reply:
@aaronsteers Issue created: https://github.com/meltano/meltano/issues/6065. Please share your thoughts in there, I want to better understand the security implications and relevance to advanced deployments.